src/main/java/com/intuit/graphql/orchestrator/federation/keydirective/KeyDirectiveValidator.java

```diff
@@ -1,31 +1,27 @@
 package com.intuit.graphql.orchestrator.federation.keydirective;
 
 import com.intuit.graphql.graphQL.Argument;
-import com.intuit.graphql.graphQL.FieldDefinition;
 import com.intuit.graphql.graphQL.InterfaceTypeDefinition;
-import com.intuit.graphql.graphQL.ObjectTypeDefinition;
 import com.intuit.graphql.graphQL.TypeDefinition;
 import com.intuit.graphql.graphQL.TypeSystemDefinition;
-import com.intuit.graphql.orchestrator.federation.keydirective.exceptions.EmptyFieldsArgumentKeyDirective;
 import com.intuit.graphql.orchestrator.federation.keydirective.exceptions.MultipleArgumentsForKeyDirective;
 import com.intuit.graphql.orchestrator.federation.keydirective.exceptions.NoFieldsArgumentForKeyDirective;
-import com.intuit.graphql.orchestrator.federation.keydirective.exceptions.InvalidKeyDirectiveFieldReference;
 import com.intuit.graphql.orchestrator.federation.keydirective.exceptions.InvalidLocationForKeyDirective;
-import org.apache.commons.collections4.CollectionUtils;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
 import org.apache.commons.lang3.StringUtils;
 
-import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
+
+import static com.intuit.graphql.orchestrator.utils.XtextUtils.checkFieldSetValidity;
 
 /**
  * This class helps break up the {@link com.intuit.graphql.orchestrator.schema.transform.KeyTransformer} by
  * validating if the TypeDefinition key directive is valid.
  */
 public class KeyDirectiveValidator {
 
-  public void validate(TypeDefinition typeDefinition, List<Argument> argumentList) {
+  public void validate(XtextGraph sourceGraph, TypeDefinition typeDefinition, List<Argument> argumentList) {
```

Comment:
we might need to refactor this to avoid passing `XtextGraph` from function to function.